### DISCUSSION TITLE
The action menu in the asset index should allow the user to delete folders

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/cms#13086] [Ideas] [answered]
When a folder is selected in the asset index listing, the `Delete` option becomes unavailable in the action menu:

![Screenshot showing the greyed-out delete option](https://user-images.githubusercontent.com/10146880/232053534-99aa1edd-9bc4-493f-973b-b230e7a6fb12.png)

The only way to delete a folder with its contents is to open it, then use the dropdown menu in the breadcrumbs nav:

![Screenshot showing the delete option for folders](https://user-images.githubusercontent.com/10146880/232053772-45083ffd-931c-4436-bd94-edca730d12c8.png)

This is unintuitive and looks like an error from a user perspective. In a typical directory browser interface, being able to select and delete a folder from the parent folder is expected and normal behaviour. The option being greyed out looks like a bug – a user will expect to be able to delete files and folders with this option. In contrast to this, having to open a folder in order to delete it is unexpected, I don't think many users will be able to find this option.

**Comment by darylknight** [comment]
+1 here. You can select Entries with the checkboxes, then use the cog wheel menu to delete them. I tried this out on a subfolder, ran into this same issue, and posted it on Discord as a bug. Please add the ability to delete subfolders from the cogwheel at the bottom.

I think one of the reasons this looks broken rather than intended is because because "Delete" _is_ actually in that menu, but greyed out - which indicates that it cannot be deleted, which is not the case.

![image](https://user-images.githubusercontent.com/1250017/233002461-135d1bad-f37c-477a-a56a-2d0e4a4ce1c5.png)

**Comment by RyanRoberts** [comment]
This caught me and a friend out too, it never occurred to me to click the arrow in the breadcrumb

**Comment by domstubbs** [comment]
I’ve just had a support call from a client about this and had I not resorted to searching through GitHub issues I’d never have figured it out. That delete does appear in the menu but greyed out says to the user “this _is_ where you delete stuff, but for unknown reasons you can’t”.

Most people are likely to give up at this point and assume there is an unresolvable feature limitation/permissions issue rather than continuing to navigate around in the hope of finding the option somewhere else.

**Comment by piotrpog** [comment]
+1 for that. seems like simple change to add.

**Comment by brandonkelly** [comment, marked as answer]
I’ve just made it possible to delete volume folders via the “Delete” asset action, for Craft 4.6. (21a4c49dcfb1fa767b9819a57d720017b056f6a4)

The action is only available when a single folder is selected. We don’t yet have the plumbing in place for multi-folder deletion.

**Reply by MoritzLost** [reply]
@brandonkelly Thanks Brandon! Being able to delete multiple files/folders at the same time would be nice, but this is already a huge improvement.

**Reply by brandonkelly** [reply]
Craft 4.6.0 is out now with that change 🎉
